backend/dataall/modules/dataset_sharing/aws/s3_client.py

```diff
@@ -121,6 +121,51 @@ def generate_access_point_policy_template(
         }
         return policy
 
+    @staticmethod
+    def generate_default_bucket_policy(
+            s3_bucket_name: str,
+            owner_roleId: list,
+            allow_owner_sid: str,
+    ):
+        policy = {
+            "Version": "2012-10-17",
+            "Statement": [
+                {
+                    "Sid": allow_owner_sid,
+                    "Effect": "Allow",
+                    "Principal": "*",
+                    "Action": "s3:*",
+                    "Resource": [
+                        f"arn:aws:s3:::{s3_bucket_name}",
+                        f"arn:aws:s3:::{s3_bucket_name}/*"
+                    ],
+                    "Condition": {
+                        "StringLike": {
+                            "aws:userId": owner_roleId
```

Comment:
Tracking it back to where this list of owner_roleIds gets generated, i can see that we are passing role arns. From this [blogpost](https://aws.amazon.com/blogs/security/how-to-restrict-amazon-s3-bucket-access-to-a-specific-iam-role/) I though that we could only add actual user ids in the policy. Have we tested that the dataset role, the environment role and the pivot role can access the bucket?

Reply:
We use the arn to only extract the role name from a role arn. We then pass it to get_role_id() which ultimately fetches the roleId. We basically return the list of Role ids for the list of arns existing within the provided aws account  number. We dont use Role arns in the policy directly.